src/resources/components/stats/stats.tsx

```diff
@@ -1,20 +1,55 @@
+import type { AccessTokenResponse, MetricsResponse } from "~types/types";
+
 const Stats = () => {
-  const statsData = {
-    hoursWatched: 254,
-    messages: 420,
-    topChannels: [
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-    ],
-    topCategories: [
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-      "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
-    ],
-  };
+  const [authorization] = useStorage<AccessTokenResponse>("accessToken");
+
+  const [stats, setStats] = useState<MetricsResponse>();
+
+  useEffect(() => {
+    getMetrics(authorization?.access_token)
+      .then((res) => {
+        console.log(res);
+        setStats(res);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }, [authorization]);
+
+  const statsData = useMemo(() => {
+    if (!stats) {
+      return {
+        hoursWatched: 420,
+        messages: 420,
+        topChannels: [
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+        ],
+        topCategories: [
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+          "https://static-cdn.jtvnw.net/jtv_user_pictures/f5c84939-a415-4654-b5da-60ff968280e6-profile_image-150x150.png",
+        ],
+      };
+    }
+    return {
+      hoursWatched: stats.main_metrics.minutes_watched
+        ? stats.main_metrics.minutes_watched * 60
+        : 0,
+      messages: 420,
+      topChannels: stats.user_metrics_by_channel.slice(0, 4).map((channel) => {
+        return `https://twitch-cdn.danielheart.dev/u/${channel.channel_id}.png`;
+      }),
+      topCategories: stats.user_metrics_by_category
+        .slice(0, 4)
+        .map((channel) => {
+          return `https://twitch-cdn.danielheart.dev/c/${channel.category_id}.png`;
+        }),
+    };
+  }, [stats]);
```

Comment:
I got this when I tried to open the stats
![image](https://github.com/user-attachments/assets/f42173a1-1786-4f08-9fed-f924916883af)

I printed `console.log(statsData.topChannels)`
![image](https://github.com/user-attachments/assets/b7cc672d-212a-4ff1-b36d-3f4b8b948d02)

Reply:
return of stats
![image](https://github.com/user-attachments/assets/f66b059d-33f2-4e31-9ff9-988a95e31e06)

Reply:
This is related to the throttle API which were fixing here: https://github.com/basementdevs/tbp-consumer-api/pull/4

Soon I'll clear the rankings and you will be able to test it again.